### DISCUSSION TITLE
Issue with the Sanity Vercel deploy plugin – shows locally, but not on Vercel

### DISCUSSION THREAD

**Discussion by sebpowell** [vercel/community#2479] [Help]
### Problem Description

I'm new to Sanity, and trying to familiarise myself with how it works.

I'm trying to work out how to trigger a build in Vercel when the content is updated in the studio. From what I gather, this requires using the following plugin: https://www.npmjs.com/package/sanity-plugin-vercel-deploy

I installed it, and it works fine locally – I can see the 'deploy' tab on `localhost:3000/studio`. However, it doesn't show up when I access the studio on Vercel.

I was wondering whether it might be a setting I need to enable somewhere, but wasn't able to see anything obvious in the docs. I've left a ticket on the project Github, but wondering if anyone here might have an idea as to where I'm going wrong.

This is my package.json:

```
{
  "name": "next-sanity-test",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "ts": "tsc --watch --noEmit"
  },
  "dependencies": {
    "@portabletext/react": "^2.0.2",
    "@sanity/block-content-to-markdown": "^0.0.5",
    "@sanity/code-input": "^4.1.1",
    "@sanity/image-url": "^1.0.2",
    "@sanity/vision": "^3.10.1",
    "@types/node": "18.15.0",
    "@types/react": "18.0.28",
    "@types/react-dom": "18.0.11",
    "eslint": "8.35.0",
    "eslint-config-next": "13.2.3",
    "next": "13.2.3",
    "next-sanity": "^4.3.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "sanity": "^3.10.1",
    "sanity-plugin-vercel-deploy": "^3.3.3",
    "typescript": "4.9.5"
  },
  "devDependencies": {
    "autoprefixer": "^10.4.14",
    "postcss": "^8.4.21",
    "prettier": "^2.8.8",
    "sanity-codegen": "^0.9.8",
    "tailwindcss": "^3.2.7"
  }
}
```

And this is my sanity.config.ts file:

```
import { visionTool } from "@sanity/vision";
import { defineConfig } from "sanity";
import { deskTool } from "sanity/desk";
import { codeInput } from "@sanity/code-input";
import { apiVersion, dataset, projectId } from "./sanity/env";
import schemas from "./sanity/schemas";
import { vercelDeployTool } from 'sanity-plugin-vercel-deploy'

export default defineConfig({
  basePath: "/studio",
  projectId,
  dataset,
  schema: {
    types: [...schemas],
  },
  plugins: [
    deskTool(),
    visionTool({ defaultApiVersion: apiVersion }),
    codeInput(),
    vercelDeployTool(),
  ],
});
```
The tab shows on localhost:

![image](https://github.com/vercel/vercel/assets/1786366/d5327932-81bd-42a5-8b66-b5fa291ba3b3)

But not on production:

![image](https://github.com/vercel/vercel/assets/1786366/02fef3a5-0725-4506-ae2d-c5720f39b63e)

The fact it's showing ok locally makes me think there might be a setting I need to enable somewhere on Vercel? Any help would be greatly appreciated.

**Comment by amyegan** [comment]
🤔 That seems to be a third-party package, so I don't know exactly how it works. It would be best to contact the maintainers of [`sanity-plugin-vercel-deploy`](https://www.npmjs.com/package/sanity-plugin-vercel-deploy) for configuration advice since they're the experts on it.

Alternatively, there is an official [Vercel Sanity integration](https://vercel.com/integrations/sanity) that might make setup easier.

**Comment by NarcOSEN** [comment]
Hello @sebpowell ,

I'm not sure if you found a solution so far for this, but I found your question while looking for a very similar issue of mine.
To explain in short, I do not have the "Deploy" button in neither local or production environments. However, my local environment would indeed update almost in real time with any changes I did, while production (hosted on Vercel) would need re-deploying.

The solution for me was to basically just follow the guide here: https://vercel.com/new/templates/next.js/blog-nextjs-sanity
This part to be more precise:

How can I remove the "Next steps" block from my blog?

You can remove it by deleting the IntroTemplate component in /components/IndexPage.tsx .

Basically I just followed the instructions in /components/IndexPage.tsx step by step and the Vercel environment started updating live with any changes I did in the studio.

Regards,
Narcis
